cpp/include/raft/comms/comms.hpp

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace raft {
+namespace comms {
+
+typedef unsigned int request_t;
+enum class datatype_t {
+  CHAR,
+  UINT8,
+  INT32,
+  UINT32,
+  INT64,
+  UINT64,
+  FLOAT32,
+  FLOAT64
+};
+enum class op_t { SUM, PROD, MIN, MAX };
+
+/**
+ * The resulting status of distributed stream synchronization
+ */
+enum class status_t {
+  commStatusSuccess,  // Synchronization successful
+  commStatusError,    // An error occured querying sync status
+  commStatusAbort     // A failure occurred in sync, queued operations aborted
+};
+
+class comms_iface {
+ public:
+  virtual ~comms_iface();
+
+  virtual int getSize() const = 0;
```

Comment:
https://github.com/rapidsai/raft/blob/branch-0.14/cpp/.clang-tidy#L143

Doesn't the above rule apply here?

Reply:
For some reason, I am not able to execute @teju85's `run-clang-tidy` script and I don't see any instructions in the codebase. I was able to run it manually, but I'm not completely sure I did it properly.

Reply:
I think it is better to enable clang-tidy in CI, now that we have it ready in cuML too?